backend/EkoFunkcje/EkoFunkcje/Features/Attachments/AddAttachmentFunction.cs

```diff
@@ -4,38 +4,49 @@
 using System.Net.Http.Headers;
 using System.Text;
 using System.Threading.Tasks;
+using System.Web.Http;
+using EkoFunkcje.Auth;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
 using Microsoft.Azure.WebJobs;
 using Microsoft.Azure.WebJobs.Extensions.Http;
 using Microsoft.Extensions.Logging;
 using Microsoft.WindowsAzure.Storage.Blob;
 
 namespace EkoFunkcje.Features.Attachments
 {
-    public static class AddAttachmentFunction
+    public class AddAttachmentFunction
     {
+        private readonly IAuth _auth;
+        public AddAttachmentFunction(IAuth auth)
+        {
+            _auth = auth;
+        }
+
         [FunctionName("AddAttachment")]
-        public static async Task<HttpResponseMessage> Run(
-            [HttpTrigger(AuthorizationLevel.Function, "post", Route = "interventions/{interventionId}/attachments")] HttpRequestMessage req,
+        public async Task<IActionResult> Run(
+            [HttpTrigger(AuthorizationLevel.Function, "post", Route = "interventions/{interventionId}/attachments")] HttpRequest req,
             [Blob("attachments/{interventionId}/{rand-guid}", FileAccess.Write , Connection = Config.StorageConnectionName)] CloudBlockBlob newBlob,
             ILogger log)
         {
-            HttpContentHeaders contentHeaders = req.Content.Headers;
-            if (contentHeaders.ContentLength == 0) {
-                return req.CreateErrorResponse(HttpStatusCode.BadRequest, "No attachment sent");
+            if (!_auth.IsAuthorized(req, "AddAttachment"))
```

Comment:
Widze, że w każdej funkcji musiałeś dodać podobny warunek. Jestem przekonany, ze to zadziała, ale zacząłem się zastanawiać, czy może nie dałoby się tego zrobić na bazie atrybutu do funkcji? :)

Niestety nie jestem ani ekspertem od C# ani od Azure Functions, ale za to znalazłem coś takiego:  https://www.c-sharpcorner.com/article/do-you-know-azure-function-have-function-filters/

Być może na bazie tego artykułu dałoby się zrobić coś podobnego i nie trzeba by modyfikowac żadnej funkcji poza dodaniem jakiegoś atrybutu, np:
```
[EkoAuthRole(Role.Empty)]
[FunctionName("AddAttachment")]
public async Task<> .... itp itd
```

Myślisz, że to mogłoby mieć sens i chciałbyś może spróbowac?
Na pewno by to pomogło w czytelności. A przy okazji unikneli byśmy zmian w ciałach funkcji - obawiam się, że mogą one spowodować jakieś niespodziewane "breaking changes" a testów żadnych automatycznych nie mamy :)

Reply:
Te filtry w tej wersji z której korzystamy zostały niestety wycofane. Zrobiłbym tak na pewno jak byłaby taka możliwość. Na ten moment nie ma jak się wbić pomiędzy wywołanie tej funkcji